internal/pkg/agent/application/upgrade/rollback.go

```diff
@@ -175,19 +179,21 @@ func restartAgent(ctx context.Context, log *logger.Logger) error {
 		}
 		log.Warnf("Failed to restart agent via control protocol: %s", err.Error())
 
-		// Next, try to restart Agent via the service.
-		log.Infof("Restarting Agent via service; attempt %d of %d", restartAttempt, maxRestartCount)
-		err = restartViaServiceFn(ctx)
-		if err == nil {
-			break
+		// Next, try to restart Agent via the service. (only if root)
+		if root {
+			log.Infof("Restarting Agent via service; attempt %d of %d", restartAttempt, maxRestartCount)
```

Comment:
Do we need to keep this in the root case, if it isn't possible in the root case?

I am worried about upgrade bugs becoming specific to the type of installation, although maybe that is unavoidable.

Reply:
This was added by @ycombinator. It think it was added to ensure that in the worse case that it would be recoverable. Sadly its just not possible to perform in non-root.